File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {
    const sums = [];
    
    for (let i = 0; i < nums.length; i++) {
        for (let j = i + 1; j < nums.length; j++) {
            for (let k = j + 1; k < nums.length; k++) {
                sums.push(nums[i] + nums[j] + nums[k]);
            }
        }
    }
    
    return sums.filter(isPrimeNumber).length;
}

function isPrimeNumber(num) {
    if (num === 1) return false;
    if (num === 2) return true;
    if (num % 2 === 0) return false;
    for (let i = 3; i <= Math.sqrt(num); i+=2) {
        if (num % i === 0) return false;
    }
    return true;
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
def solution(nums):
    def isPrimeNumber(num):
        if num <= 1 or num % 2 == 0:
            return False
        elif num == 2:
            return True
        else:
            for i in range(3, int(num ** 0.5) + 1, 2):
                if num % i == 0:
                    return False
            return True
    
    results = []
    visited = [False] * len(nums)
    
    def dfs(sum, depth, start_idx):
        if depth == 3:
            if isPrimeNumber(sum):
                results.append(sum)
            return
        
        for i in range(start_idx, len(nums)):
            if not visited[i]:
                visited[i] = True
                dfs(sum + nums[i], depth + 1, i + 1)
                visited[i] = False
        
    
    dfs(0, 0, 0)
    
    return len(results)